android/src/main/java/com/epicshaggy/biometric/AuthSignActivity.java

```diff
@@ -0,0 +1,165 @@
+                    .setTitle(getIntent().hasExtra("title") ? getIntent().getStringExtra("title") : "Authenticate")
+                    .setSubtitle(getIntent().hasExtra("subtitle") ? getIntent().getStringExtra("subtitle") : null)
+                    .setDescription(getIntent().hasExtra("description") ? getIntent().getStringExtra("description") : null);
+
+            boolean useFallback = getIntent().getBooleanExtra("useFallback", false);
+
+            if (useFallback) {
+                builder.setAllowedAuthenticators(BIOMETRIC_STRONG | DEVICE_CREDENTIAL);
+            } else {
+                builder.setNegativeButtonText(getIntent().hasExtra("negativeButtonText") ? getIntent().getStringExtra("negativeButtonText") : "Cancel");
+            }
+
+            BiometricPrompt.PromptInfo promptInfo = builder.build();
+
+            BiometricPrompt biometricPrompt = new BiometricPrompt(this, executor, new BiometricPrompt.AuthenticationCallback() {
+                @Override
+                public void onAuthenticationError(int errorCode, @NonNull CharSequence errString) {
+                    super.onAuthenticationError(errorCode, errString);
+                    finishActivity("authenticationError", "", errorCode);
+                }
+
+                @Override
+                public void onAuthenticationSucceeded(@NonNull BiometricPrompt.AuthenticationResult result) {
+                    super.onAuthenticationSucceeded(result);
+                    String signedString = null;
+
+                    try {
+                        BiometricPrompt.CryptoObject cryptoObject = result.getCryptoObject();
+                        Signature cryptoSignature = cryptoObject.getSignature();
+                        Charset charset = StandardCharsets.UTF_8;
+                        byte[] challengeData = getIntent().getStringExtra("challengeString").getBytes(charset);
+                        cryptoSignature.update(challengeData);
+                        byte[] signed = cryptoSignature.sign();
+                        signedString = Base64.encodeToString(signed, Base64.DEFAULT);
+                    } catch (Exception e) {
+                        e.printStackTrace();
+                    }
+
+                    finishActivity("success", signedString, 0);
+                }
+
+                @Override
+                public void onAuthenticationFailed() {
+                    super.onAuthenticationFailed();
+                    counter++;
+                    if (counter == MAX_ATTEMPTS)
+                        finishActivity("failed", "", 0);
+                }
+            });
+
+            biometricPrompt.authenticate(promptInfo, cryptoObject);
+
+        } catch (UserNotAuthenticatedException e) {
+            e.printStackTrace();
+        } catch (KeyPermanentlyInvalidatedException e) {
+            e.printStackTrace();
+            finishActivity("biometryChanged", "", 666);
```

Comment:
Dumb question, what 666 means here?

Reply:
666 means biometry changed error, I also made it up (lol) so it's not conflicting with the current android mapped error. Any suggestion for the error code mas?

Reply:
It's okay to use this number haha. I thought it was a very specific coincidence that biometric changed error is using 666